backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/purchasecontract/addedvalue/interim/AVInterimComputingMethod.java

```diff
@@ -22,32 +22,163 @@
+	private final IOrderLineBL orderLineBL = Services.get(IOrderLineBL.class);
+	private final IFlatrateBL flatrateBL = Services.get(IFlatrateBL.class);
+	@NonNull private final ShippingNotificationRepository shippingNotificationRepository;
+	@NonNull private final ModularContractLogService modularContractLogService;
+	@NonNull private final MoneyService moneyService;
+
 	public AVInterimComputingMethod(
 			@NonNull final ShippingNotificationRepository shippingNotificationRepository,
 			@NonNull final ModularContractProvider contractProvider,
 			@NonNull final ModCntrInvoicingGroupRepository invoicingGroupRepository,
 			@NonNull final ModularContractLogService modularContractLogService,
-			@NonNull final ModularLogInterestRepository modularLogInterestRepository)
+			@NonNull final ModularLogInterestRepository modularLogInterestRepository,
+			@NonNull final MoneyService moneyService)
 	{
 		super(shippingNotificationRepository, contractProvider, invoicingGroupRepository, modularContractLogService, modularLogInterestRepository);
+		this.shippingNotificationRepository = shippingNotificationRepository;
+		this.modularContractLogService = modularContractLogService;
+		this.moneyService = moneyService;
+
 	}
 
 	@Override
 	public @NonNull ComputingMethodType getComputingMethodType()
 	{
 		return ComputingMethodType.AddValueOnInterim;
 	}
+
+	@Override
+	public boolean applies(final @NonNull TableRecordReference recordRef, @NonNull final LogEntryContractType logEntryContractType)
+	{
+		if (!logEntryContractType.isModularContractType())
+		{
+			return false;
+		}
+		if (recordRef.tableNameEqualsTo(I_C_Flatrate_Term.Table_Name))
+		{
+			final I_C_Flatrate_Term flatrateTermRecord = flatrateBL.getById(FlatrateTermId.ofRepoId(recordRef.getRecord_ID()));
+			if (!TypeConditions.ofCode(flatrateTermRecord.getType_Conditions()).isInterimContractType())
+			{
+				return false;
+			}
+
+			final OrderLineId orderLineId = OrderLineId.ofRepoIdOrNull(flatrateTermRecord.getC_OrderLine_Term_ID());
+			if (orderLineId == null)
+			{
+				return false;
+			}
+
+			final OrderId orderId = orderLineBL.getOrderIdByOrderLineId(orderLineId);
+			return SOTrx.ofBoolean(orderBL.getById(orderId).isSOTrx()).isPurchase();
+		}
+		if (recordRef.tableNameEqualsTo(I_M_Shipping_NotificationLine.Table_Name))
+		{
+			final I_M_Shipping_NotificationLine line = shippingNotificationRepository.getLineRecordByLineId(ShippingNotificationLineId.ofRepoId(recordRef.getRecord_ID()));
+
+			final I_C_Order salesOrder = orderBL.getById(OrderId.ofRepoId(line.getC_Order_ID()));
+			final YearAndCalendarId yearAndCalendarId = YearAndCalendarId.ofRepoIdOrNull(salesOrder.getHarvesting_Year_ID(), salesOrder.getC_Harvesting_Calendar_ID());
+
+			return yearAndCalendarId != null;
+		}
+
+		return false;
+	}
+
+	@Override
+	protected void splitLogsIfNeeded(final Money reconciledAmount, final AtomicReference<ModularContractLogEntryId> initialInterimContractId)
```

Comment:
as above, drop the AtomicReference and just pass in a nullable ModularContractLogEntryId